### PR TITLE
RabbitMQ configuration

### PR DESCRIPTION
**Pull request by HorstmannA**
Move the RabbitMQ configuration script to the python-zocalo repo.
Use site config for the configuration files path, and a command-line argument for the RabbitMQ configuration file name.

### CODE REVIEW

File: src/zocalo/cli/configure-rabbitmq.py [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+    for group in yaml_data["groups"]:
+        if group.get("settings", {}).get("broadcast"):
+            exchange_specs.extend(get_exchange_specs(group))
+        else:
+            queue_specs.extend(get_queue_specs(group))
+            binding_specs.extend(get_binding_specs(group))
+
+    policies = get_policy_specs(yaml_data["policies"])
+
+    # import re
```
Comment (d-j-hatton) [inline src/zocalo/cli/configure-rabbitmq.py:264]: I think this commented out stuff can go

**Comment by d-j-hatton** [comment]
I suspect the `hash_password` stuff is dealt with somewhere in `util.rabbitmq.py` but will have to take a closer look to see how to use it

File: src/zocalo/cli/configure-rabbitmq.py [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        if ic not in current:
+            logger.info(f"creating {cls.__name__} {ic}")
+            api.create_component(ic)
+
+
+def hash_password(passwd: str) -> str:
+    salt = os.urandom(4)
+    utf8 = passwd.encode("utf-8")
+    temp1 = salt + utf8
+    temp2 = hashlib.sha256(temp1).digest()
```
Comment (Anthchirp) [inline src/zocalo/cli/configure-rabbitmq.py:120]:
```suggestion
    temp2 = hashlib.sha256(temp1).digest()  # lgtm
```
The LGTM crypto warning is correct, but also nothing we can do about it. Other than adding a comment to the offending line.

**Review by Anthchirp — APPROVED** [review]
Need to add the zocalo.configure-rabbitmq command (assuming this is its intended name) to the `console_scripts` and `libtbx.dispatcher.script` entry points in `/setup.cfg`.

File: src/zocalo/cli/configure-rabbitmq.py [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+    BindingSpec,
+    DestinationType,
+    ExchangeSpec,
+    PolicySpec,
+    QueueSpec,
+    UserSpec,
+)
+
+
+logger = logging.getLogger("rabbitmq/configure_rabbitmq")
```
Comment (Anthchirp) [inline src/zocalo/cli/configure-rabbitmq.py:25]:
```suggestion
logger = logging.getLogger("zocalo.cli.configure_rabbitmq")
```

**Pull request merged**